integration-tests/src/tests/client/chunks_management.rs

```diff
@@ -373,101 +371,83 @@ fn chunks_recovered_from_others() {
-            );
-        })
-    });
+    Test {
+        validator_groups: 4,
+        chunk_only_producers: false,
+        drop_to_4_from: &["test1"],
+        drop_all_chunk_forward_msgs: true,
+        block_timeout: 2 * CHUNK_REQUEST_SWITCH_TO_FULL_FETCH_MS,
+    }
+    .run()
 }
 
 // And now let's add chunk-only producers into the mix
 
 /// Happy case -- each shard is handled by one cop and one block producers.
 #[test]
 fn chunks_produced_and_distributed_one_val_shard_cop() {
-    heavy_test(|| {
-        run_actix(async {
-            chunks_produced_and_distributed_common(
-                4,
-                true,
-                &[],
-                false,
-                15 * CHUNK_REQUEST_RETRY_MS,
-            );
-        });
-    });
+    Test {
+        validator_groups: 4,
+        chunk_only_producers: true,
+        drop_to_4_from: &[],
+        drop_all_chunk_forward_msgs: false,
+        block_timeout: 15 * CHUNK_REQUEST_RETRY_MS,
+    }
+    .run()
 }
 
 /// `test4` can't talk to `test1`, so it'll fetch the chunk for first shard from `cop1`.
 #[test]
 fn chunks_recovered_from_others_cop() {
-    heavy_test(|| {
-        run_actix(async {
-            chunks_produced_and_distributed_common(
-                1,
-                true,
-                &["test1"],
-                true,
-                4 * CHUNK_REQUEST_SWITCH_TO_OTHERS_MS,
-            );
-        });
-    });
+    Test {
+        validator_groups: 1,
+        chunk_only_producers: true,
+        drop_to_4_from: &["test1"],
+        drop_all_chunk_forward_msgs: true,
+        block_timeout: 4 * CHUNK_REQUEST_SWITCH_TO_OTHERS_MS,
+    }
+    .run()
 }
 
 /// `test4` can't talk neither to `cop1` nor to `test1`, so it can't fetch chunk
 /// from chunk producers and has to reconstruct it.
 #[test]
 #[should_panic]
 fn chunks_recovered_from_full_timeout_too_short_cop() {
-    heavy_test(|| {
-        run_actix(async {
-            chunks_produced_and_distributed_common(
-                4,
-                true,
-                &["test1", "cop1"],
-                true,
-                2 * CHUNK_REQUEST_SWITCH_TO_OTHERS_MS,
-            );
-        });
-    });
+    Test {
+        validator_groups: 4,
+        chunk_only_producers: true,
+        drop_to_4_from: &["test1", "cop1"],
+        drop_all_chunk_forward_msgs: true,
+        block_timeout: 2 * CHUNK_REQUEST_SWITCH_TO_OTHERS_MS,
+    }
+    .run()
 }
 
 /// Same as above, but with longer block production timeout which should allow for full reconstruction.
 #[test]
 fn chunks_recovered_from_full_cop() {
-    heavy_test(|| {
-        run_actix(async {
-            chunks_produced_and_distributed_common(
-                4,
-                false,
-                &["test1", "cop1"],
-                true,
-                2 * CHUNK_REQUEST_SWITCH_TO_FULL_FETCH_MS,
-            );
-        })
-    });
+    Test {
+        validator_groups: 4,
+        chunk_only_producers: true,
```

Comment:
Huh, this is an unintended bugfix I think, this would should be `true`

Reply:
I'm not sure if chunk_only_producers would be true here. The naming for the rest of the tests seem to indicate that if chunk_only_producers is true, the name is the tests would end with _cop

Reply:
And this test does end with `_cop`

Reply:
Ie, it was wrongly `false` before.

Reply:
ah sorry I must have misread something